### DISCUSSION TITLE
Support for custom Kerberos Ccache File Path - Database Hooks

### DISCUSSION THREAD

**Discussion by VA0** [apache/airflow#32427] [Ideas]
Hello,

We would like to have an option to specify per ex. odbc / mssql connection a custom path for kerberos cache file.
The reason is we use different user and realms and the default principal in klist is not correct for the 2nd and 3th connection.
Unfortunately our company forces us to use only windows authentication via kerberos so cant really use simple sql login user.

Do you have any temporary thoughts of how to deal with that issue? I mean i can write kinit in the dag and change the default principal but then it interfere with our dags.

Thank you.

**Comment by potiuk** [comment]
You can extend (or copy) the operators that use different paths based on different variables or whatever you chose and make custom versions of them. Creating custom operators is first-class citizen in Airflow, and since this is a simple Python code it's as easy as wrting DAGS. The only thing is that you will have to refresh all of those separately, `airflow kerberos` command will refresh only the single one that Airflow supports. But this can be done independently from Airflow.
